### PR TITLE
Add Camera Component to Component Shelf, Camera Prefab to Entity Shelf

### PR DESCRIPTION
**Pull request by MbfloydIR**
renamed the folder for the camera properties
created node editor for the camera component
added to/from json code in the camera component
added camera component to the ui shelf
added camera prefab
https://tsu.atlassian.net/browse/IR-3374
## References
closes #_insert number here_

### CODE REVIEW

File: packages/ui/src/components/editor/properties/camera/index.tsx
Comment (dinomut1) [inline packages/ui/src/components/editor/properties/camera/index.tsx:1]: We need to be able to remove this component. Please wrap this component in a `<NodeEditor>` element. You can use the model component's index.tsx as a reference

**Review by dinomut1 — DISMISSED** [review]

File: packages/ui/src/components/editor/properties/camera/index.tsx
Comment (MbfloydIR) [reply packages/ui/src/components/editor/properties/camera/index.tsx:1]: implemented

**Review by SYBIOTE — APPROVED** [review]

File: packages/spatial/src/camera/components/CameraComponent.ts [hunk 1]
```diff
@@ -40,13 +41,23 @@ export const CameraComponent = defineComponent({
     camera.cameras = [new PerspectiveCamera().copy(camera, false)]
     return camera
   },
-  onSet: (entity, component, json: undefined) => {
+  onSet: (entity, component, json) => {
     addObjectToGroup(entity, component.value as ArrayCamera)
```
Comment (SYBIOTE) [inline packages/spatial/src/camera/components/CameraComponent.ts:45]: please add object in a useEffect not in onSet

**Review by SYBIOTE — CHANGES_REQUESTED** [review]

File: packages/spatial/src/camera/components/CameraComponent.ts (hunk 1 above)
Comment (MbfloydIR) [reply packages/spatial/src/camera/components/CameraComponent.ts:45]: implemented

**Review by SYBIOTE — APPROVED** [review]

**Pull request merged**